driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -0,0 +1,430 @@
+        }
+
+        cursor = collection.watch().cursor();
+        try {
+            assertNull(cursor.getResumeToken());
+        } finally {
+            cursor.close();
+        }
+    }
+
+    //
+    // 14. For a ChangeStream under these conditions:
+    //   The batch is not empty.
+    //   The batch hasn’t been iterated at all.
+    //   Only the initial aggregate command has been executed.
+    // Expected result:
+    //   getResumeToken must return resumeAfter from the initial aggregate if the option was specified.
+    //   If resumeAfter is not specified, the getResumeToken result must be empty.
+    //
+    @Test
+    public void testGetResumeTokenReturnsResumeAfter() {
+        BsonDocument resumeAfterResumeToken;
+        MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = collection.watch().cursor();
+        collection.insertOne(Document.parse("{ _id: 42, x: 1 }"));
+        collection.insertOne(Document.parse("{ _id: 43, x: 1 }"));
+        try {
+            resumeAfterResumeToken = cursor.next().getResumeToken();
+        } finally {
+            cursor.close();
+        }
+
+        cursor = collection.watch().resumeAfter(resumeAfterResumeToken).cursor();
+        try {
+            assertEquals(resumeAfterResumeToken, cursor.getResumeToken());
+        } finally {
+            cursor.close();
+        }
+
+        cursor = collection.watch().cursor();
+        try {
+            assertNull(cursor.getResumeToken());
+        } finally {
+            cursor.close();
+        }
+    }
+
+    //
+    // 15. For a ChangeStream under these conditions:
+    //   Running against a server >=4.0.7.
+    //   The batch is not empty.
+    //   The batch hasn’t been iterated at all.
+    //   The stream has iterated beyond a previous batch and a getMore command has just been executed.
+    // Expected result:
+    //   getResumeToken must return the postBatchResumeToken from the previous command response.
+    //
+    @Test
+    public void testGetResumeTokenReturnsPostBatchResumeTokenAfterGetMore()
+            throws NoSuchFieldException, IllegalAccessException {
+        assumeTrue(serverVersionAtLeast(asList(4, 0, 7)));
+
+        MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor = collection.watch().cursor();
+        collection.insertOne(Document.parse("{ _id: 42, x: 1 }"));
+        try {
+            // use reflection to access the postBatchResumeToken
+            BatchCursor<?> batchCursor = getBatchCursor(cursor);
+
+            // check equality in the case where the batch has not been iterated at all
+            assertEquals(cursor.getResumeToken(), batchCursor.getPostBatchResumeToken());
+
+            cursor.next();
+            assertEquals(cursor.getResumeToken(), batchCursor.getPostBatchResumeToken());
+        } finally {
+            cursor.close();
+        }
+    }
+
+    private BatchCursor<?> getBatchCursor(final MongoChangeStreamCursor<ChangeStreamDocument<Document>> cursor)
```

Comment:
Maybe it's better to move this helper method at the end of this file?

Reply:
I have moved the method to the end of the file.